packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -243,11 +268,25 @@ function hydrateCustomElement(
     vnode: VCustomElement,
     renderer: RendererAPI
 ): Node | null {
-    if (
-        !hasCorrectNodeType<Element>(vnode, elm, EnvNodeTypes.ELEMENT, renderer) ||
-        !isMatchingElement(vnode, elm, renderer)
-    ) {
-        return handleMismatch(elm, vnode, renderer);
+    const { validationOptOut } = vnode.ctor;
+    const attrsToNotValidate = getAttrsToNotValidate(validationOptOut);
```

Comment:
I think the rest of this code could be a bit simpler if `getAttrsToNotValidate` returned a function that takes a string as input and outputs a boolean. In the case where `validationOptOut` is falsey, it would just always return true.

There might be a slight perf overhead, but I wouldn't worry about this too much until we have a profiler telling us that it's a problem. Then we can get rid of the `if (!validationOptOut || attrsToNotValidate) {`, the question marks in `attrsToNotValidate?.has?.('class')`, etc.